pkgs/tools/virtualization/google-compute-engine/default.nix

```diff
@@ -0,0 +1,55 @@
+{ lib
+, fetchFromGitHub
+, pythonPackages
+, hwclock
+, shadow
+, systemd
+, utillinux
+}:
+let
+  version = "20170523";
+in
+pythonPackages.buildPythonApplication {
+  name = "google-compute-engine-${version}";
+  namePrefix = "";
+
+  src = fetchFromGitHub {
+    owner = "GoogleCloudPlatform";
+    repo = "compute-image-packages";
+    rev = version;
+    sha256 = "1qxyj3lj9in6m8yi6y6wcmc3662h9z4qax07v97rdnay99mxdv68";
+  };
+
+  patches = [ ./0001-allow-nologin-other-paths.patch ];
+
+  patchPhase = ''
+    for file in $(find google_compute_engine -type f); do
+      substituteInPlace "$file" \
+        --replace /bin/systemctl "${systemd}/bin/systemctl" \
+        --replace /bin/bash "${bash}/bin/bash" \
+        --replace /sbin/hwclock "${hwclock}/bin/hwclock"
+
+      # SELinux tool ???  /sbin/restorecon
+    done
+
+    substituteInPlace google_config/udev/65-gce-disk-naming.rules \
+      --replace /bin/sh "${bash}/bin/sh" \
+      --replace /bin/umount "${utillinux}/bin/umount" \
+      --replace /usr/bin/logger "${utillinux}/bin/logger"
+  '';
+
+  postInstall = ''
+    # allows to install the package in `services.udev.packages` in NixOS
+    mkdir -p $out/lib/udev/rules.d
+    cp -r google_config/udev/*.rules $out/lib/udev/rules.d
+  '';
+
+  propagatedBuildInputs = with pythonPackages; [ boto setuptools ];
```

Comment:
Apparently just for one thing: setup boto to use the Google Cloud Storage as a backend (GCS has a S3-compatible API) ^_^ https://github.com/GoogleCloudPlatform/compute-image-packages#instance-setup